misk/src/main/kotlin/misk/hibernate/Session.kt

```diff
@@ -1,11 +1,20 @@
 package misk.hibernate
 
+import java.sql.Connection
 import kotlin.reflect.KClass
 
 interface Session {
   val hibernateSession: org.hibernate.Session
   fun <T : DbEntity<T>> save(entity: T): Id<T>
   fun <T : DbEntity<T>> load(id: Id<T>, type: KClass<T>): T
+  fun shards(): List<Shard>
+  fun <T> target(shard: Shard, function: () -> T): T
+  fun <T> doWork(work: (Connection) -> T): T
```

Comment:
should we give this a nicer name? This is what Hibernate uses but it never made much sense to me

Reply:
I changed it to `useConnection` which I'd argue is one step up from `doWork` but far from perfect. The main issue is that the word `use` seems to imply that the connection is closed afterwards?